packages/ra-ui-materialui/src/detail/Tab.tsx

```diff
@@ -73,7 +74,9 @@ const Tab = ({
             value={value}
             icon={icon}
             className={classnames('show-tab', className)}
-            {...({ component: Link, to: value } as any)} // to avoid TypeScript screams, see https://github.com/mui-org/material-ui/issues/9106#issuecomment-451270521
+            {...(syncWithLocation
```

Comment:
I find it weird that you handle the tab click here when `syncWithLocation` is true, and in the parent component when it's false.

I think you'd better move the tab click handling in the parent component altogether.

Same in formTab.

Reply:
That's because it's a simple link when `syncWithLocation` is `true`. I'd rather avoid having buttons calling the router methods as it is less accessible imo